src/topp/MzTabExporter.cpp

```diff
@@ -120,6 +120,83 @@ namespace OpenMS
       setValidFormats_("out", ListUtils::create<String>("tsv"));
     }
 
+    /// Helper:
+    /// gets peptide_evidences with data from internal structures,
+    /// a row with information about the PeptideHit in general
+    /// a reference to a vector of rows
+    /// and adds a row for each PeptideEvidence to rows.
+    static void addPepEvidenceToRows(const vector<PeptideEvidence>& peptide_evidences, MzTabPSMSectionRow& row, MzTabPSMSectionRows& rows)
+    {
+      if (!peptide_evidences.empty())
+      {
+        for (Size i = 0; i != peptide_evidences.size(); ++i)
+        {
+          // get AABefore and AAAfter as well as start and end for all pep evidences
+
+          // pre/post
+          // from spec: Amino acid preceding the peptide (coming from the PSM) in the protein
+          // sequence. If unknown “null” MUST be used, if the peptide is N-terminal “-“
+          // MUST be used.
+          if (peptide_evidences[i].getAABefore() == PeptideEvidence::UNKNOWN_AA)
+          {
+            row.pre = MzTabString("null");
+          }
+          else if (peptide_evidences[i].getAABefore() == PeptideEvidence::N_TERMINAL_AA)
+          {
+            row.pre = MzTabString("-");
+          }
+          else
+          {
+            row.pre = MzTabString(String(peptide_evidences[i].getAABefore()));
+          }
+
+          if (peptide_evidences[i].getAAAfter() == PeptideEvidence::UNKNOWN_AA)
+          {
+            row.post = MzTabString("null");
+          }
+          else if (peptide_evidences[i].getAAAfter() == PeptideEvidence::C_TERMINAL_AA)
+          {
+            row.post = MzTabString("-");
+          }
+          else
+          {
+            row.post = MzTabString(String(peptide_evidences[i].getAAAfter()));
+          }
+
+          // start/end
+          if (peptide_evidences[i].getStart() == PeptideEvidence::UNKNOWN_POSITION)
+          {
+            row.start = MzTabString("-");
+          }
+          else
+          {
+            row.start = MzTabString(String(peptide_evidences[i].getStart() + 1)); // counting in mzTab starts at 1
+          }
+
+          if (peptide_evidences[i].getEnd() == PeptideEvidence::UNKNOWN_POSITION)
+          {
+            row.end = MzTabString("-");
+          }
+          else
+          {
+            row.end = MzTabString(String(peptide_evidences[i].getEnd() + 1)); // counting in mzTab starts at 1
+          }
+
+          row.accession = MzTabString(peptide_evidences[i].getProteinAccession());
+
+          rows.push_back(row);
+        }
+      }
+      else
+      { // report without pep evidence information
+        row.pre = MzTabString("-");
```

Comment:
just checked it:
Amino acid preceding the peptide (coming from the PSM) in the protein
sequence. If unknown “null” MUST be used, if the peptide is N-terminal “-“
MUST be used. 
So - is wrong